# 과제: CESM 산업화 이전 제어 시뮬레이션에서의 전 지구 평균 수지

이 노트북은 알바니 대학교의 [브라이언 E. J. 로즈](Brian E. J. Rose) 교수가 저술한 [기후 실험실](The Climate Laboratory)의 일부입니다.

## 학습 목표

이 과제를 완수하는 학생들은 다음 기술과 개념을 습득하게 될 것입니다:

- Jupyter 노트북 사용에 대한 지속적인 연습
- CESM 시뮬레이션의 대기 출력물에 대한 이해
- CESM 제어 시뮬레이션의 전지구 에너지 수지를 관측 자료와 더욱 완전하게 비교
- 표면 온도의 연간 주기를 관측 자료와 비교하여 검증
- 이러한 전지구 온도 변화에 대한 가설을 수립할 기회
- Python 프로그래밍 기술: 기본적인 xarray 사용법 (격자형 데이터셋(gridded dataset) 열기 및 평균 계산)

## 지침

- 이 노트북의 로컬 사본(JupyterHub 또는 개인 장치에서)에 **추가 셀에 답변을 작성하세요**.
- 아래의 **필수 문제들을 완료하세요**.
- 텍스트를 작성할 때는 셀 유형을 `Markdown`으로, Python 코드를 작성할 때는 `Code`로 설정하는 것을 잊지 마세요!
- 필요한 경우 코드에 여러분의 방법을 설명하는 **주석을 포함하세요**.
- 질문에 실제로 답변하는 것을 잊지 마세요. (코드와 그림만으로는 불충분하며) **서면 답변이 필수입니다**!
- 여러분의 텍스트, 코드, 그림을 모두 포함하는 **단일 Jupyter 노트북** 파일로 솔루션을 제출하세요.
- *여러분의 노트북이* ***오류 없이 깔끔하게 실행되는지 확인하세요:***
    - 노트북을 저장하세요
    - `Kernel` 메뉴에서 `Restart & Run All`을 선택하세요
    - 노트북이 처음부터 끝까지 오류 없이 실행되어 예상되는 결과물을 생성했나요?
    - 그렇다면, 다시 저장하고 노트북 파일을 제출하세요
    - 그렇지 않다면, 오류를 수정하고 다시 시도하세요.

## 문제 1: CESM 제어 시뮬레이션의 전 지구 에너지 수지

다음 각 양에 대해 **전 지구 평균, 시간 평균** 값을 계산하고, 이를 강의 자료에 있는 Trenberth and Fasullo (2012) 그림의 관측값과 비교하세요.

- 태양 복사 (단파 복사) 수지:
    - 입사 태양 복사 (Incoming Solar Radiation), 또는 일사량 (Insolation)
    - 대기권 최상단에서 반사된 태양 복사
    - 지표면에 의해 반사된 태양 복사
    - 지표면에 흡수된 태양 복사
    - 구름과 대기에 의해 반사된 태양 복사 *(이 값은 대기권 최상단에서 반사된 복사량과 지표면에서 반사된 복사량의 차이로 계산할 수 있습니다)*
    - 대기권 최상단에서 총 흡수된 태양 복사 (ASR)
    - 대기에 흡수된 태양 복사 *(이 값은 수지의 잔여량, 즉 다른 모든 흡수 및 반사량을 고려한 후 남은 양으로 계산할 수 있습니다)*
- 장파 복사 수지:
    - 외향 장파 복사 (Outgoing Longwave Radiation)
    - 지표면에서 위로 방출되는 복사
    - 지표면으로 향하는 하향 복사
- 기타 지표면 플럭스:
    - "열(Thermals)", 또는 *현열 플럭스 (sensible heat flux)*. *이 값은 데이터셋에서 `SHFLX`라는 필드에서 찾을 수 있습니다.*
    - "증발산(Evapotranspiration)", 또는 *잠열 플럭스 (latent heat flux)*. *이 값은 데이터셋에서 `LHFLX`라는 필드에서 찾을 수 있습니다.*

*참고: 대기 흡수 및 방출 과정에 대해서는 나중에 더 자세히 살펴보겠습니다. "대기 방출(Emitted by Atmosphere)" 또는 "대기창(Atmospheric Window)"과 같은 항을 계산하려고 시도할 필요는 없습니다*

**위 결과를 바탕으로 다음 질문에 답하세요:**

- CESM 제어 시뮬레이션은 **에너지 균형 (energy balance)** 상태에 (또는 그 근처에) 있습니까? 이것이 모델링된 기후가 평형 상태에 가까운지 여부에 대해 무엇을 시사합니까?
- CESM 시뮬레이션의 전 지구 에너지 수지와 관측값 간의 가장 중요한 유사점과 차이점이 무엇이라고 생각하는지 자신의 말로 요약하세요.

## 문제 2: 관측치와 비교하여 전 지구 평균 지표면 온도의 연간 주기 검증

수업 자료에서 우리는 CESM 제어 시뮬레이션의 **전 지구 평균 지표면 온도 시계열**을 플로팅하여 **연간 주기**를 발견했습니다. 이 실습의 목적은 이 현상이 관측된 온도 기록에서도 나타나는지 확인하는 것입니다. 만약 그렇다면, 우리는 이것이 수치 모델의 인공물이 아니라 지구 기후의 실제 특징이라고 결론 내릴 수 있습니다.

관측 자료로는 **NCEP 재분석 데이터(NCEP Reanalysis data)**를 사용할 것입니다.

*재분석 데이터(Reanalysis data)는 실제로 관측 자료와 수치 일기 예보 모델의 출력 결과를 혼합한 것입니다. 이는 관측 자료가 매우 희박한 지역을 포함하여 전 지구의 상태에 대한 "최고의 추정치"를 나타냅니다.*

필요한 데이터는 모두 인터넷을 통해 제공됩니다. 우리는 1981년부터 2010년까지 30년 기간 동안 평균된 월별 기후값(monthly climatologies)을 살펴볼 것입니다.

사용 가능한 데이터는 다음에서 찾아볼 수 있습니다:
https://www.esrl.noaa.gov/psd/data/gridded/data.ncep.reanalysis.derived.html

**지표면 기온(Surface air temperature)**은 `air.2m.mon.ltm.nc`라는 파일에 포함되어 있으며, 이 파일은 `Surface Fluxes`라는 컬렉션에서 찾을 수 있습니다.

이 데이터 파일의 카탈로그 페이지로 바로 가는 링크입니다:
https://www.esrl.noaa.gov/psd/thredds/catalog/Datasets/ncep.reanalysis.derived/surface_gauss/catalog.html?dataset=Datasets/ncep.reanalysis.derived/surface_gauss/air.2m.day.ltm.nc

이제 `OPeNDAP` 링크를 클릭하세요. 파일 내용에 대한 많은 정보가 담긴 페이지가 열릴 것입니다. `Data URL`은 데이터를 파이썬 세션으로 읽어오는 데 필요한 것입니다. 예를 들어, 다음 코드는 파일을 열고 파일에 포함된 변수 목록을 표시합니다:
```python
import xarray as xr
ds = xr.open_dataset('https://www.esrl.noaa.gov/psd/thredds/dodsC/Datasets/ncep.reanalysis.derived/surface_gauss/air.2m.day.ltm.nc')
print(ds)
```

In [1]:
import xarray as xr
url = "http://www.esrl.noaa.gov/psd/thredds/dodsC/Datasets/ncep.reanalysis.derived/surface_gauss/air.2m.mon.ltm.nc"
ncep_air2m = xr.open_dataset(url, decode_times=False)

ncep_air2m

<xarray.Dataset> Size: 2MB
Dimensions:             (lat: 94, lon: 192, time: 12, nbnds: 2)
Coordinates:
  * lat                 (lat) float32 376B 88.54 86.65 84.75 ... -86.65 -88.54
  * lon                 (lon) float32 768B 0.0 1.875 3.75 ... 354.4 356.2 358.1
  * time                (time) float64 96B -1.577e+07 -1.577e+07 ... -1.576e+07
Dimensions without coordinates: nbnds
Data variables:
    climatology_bounds  (time, nbnds) float64 192B ...
    air                 (time, lat, lon) float32 866kB ...
    valid_yr_count      (time, lat, lon) float32 866kB ...
Attributes:
    Conventions:                    COARDS
    description:                    Data is from NMC initialized reanalysis\n...
    platform:                       Model
    NCO:                            20121012
    history:                        Created 2021/07/26 by doMonthLTMNC4
    title:                          monthly mean air.2m from the NCEP Reanalysis
    dataset_title:                  NCEP-NCAR Reanalysis 1
    References:                     http://www.psl.noaa.gov/data/gridded/data...
    not_missing_threshold_percent:  minimum 3% values input to have non-missi...
    _NCProperties:                  version=2,netcdf=4.7.0,hdf5=1.10.5,

온도 데이터는 `air`라고 불립니다. 자세한 내용을 살펴보세요:

In [2]:
ncep_air2m.air

<xarray.DataArray 'air' (time: 12, lat: 94, lon: 192)> Size: 866kB
[216576 values with dtype=float32]
Coordinates:
  * lat      (lat) float32 376B 88.54 86.65 84.75 82.85 ... -84.75 -86.65 -88.54
  * lon      (lon) float32 768B 0.0 1.875 3.75 5.625 ... 352.5 354.4 356.2 358.1
  * time     (time) float64 96B -1.577e+07 -1.577e+07 ... -1.576e+07 -1.576e+07
Attributes: (12/13)
    long_name:     Long Term Mean Monthly Mean of Air Temperature
    valid_range:   [150. 400.]
    units:         degK
    precision:     2
    GRIB_id:       11
    GRIB_name:     TMP
    ...            ...
    level_desc:    2 m
    statistic:     Long Term Mean
    parent_stat:   Mean
    dataset:       NCEP Reanalysis Derived Products
    actual_range:  [199.70786 312.07498]
    _ChunkSizes:   [  1  94 192]

차원은 `(time: 12, lat: 94, lon: 192)`임을 주목하세요. 시간 차원은 월 단위입니다. 하지만 위도/경도(lat/lon) 격자는 우리 모델 출력물과 같지 않음에 유의하세요!

*이 데이터의 전 지구 평균을 어떻게 계산할지 생각해 보세요.*

### 당신의 과제:

- 전지구 평균 관측 지표면 기온 기후값(climatology)에 대해 레이블이 명확한 시계열 그래프를 작성하세요.
- CESM 시뮬레이션에서 발견한 연간 주기(annual cycle)가 관측값에서도 존재하는지 확인하세요.
- 이 연간 주기(annual cycle)가 존재하는 이유에 대한 타당한 물리적 설명을 본인의 언어로 제시하세요.

## 크레딧

이 노트북은 올버니 대학교(University at Albany)의 [브라이언 E. J. 로즈(Brian E. J. Rose)](https://www.atmos.albany.edu/facstaff/brose/)가 개발하고 유지 관리하는 오픈 소스 교재인 [더 클라이밋 래버러토리(The Climate Laboratory)](https://brian-rose.github.io/ClimateLaboratoryBook)의 일부입니다.

이 자료는 [크리에이티브 커먼즈 저작자표시 4.0 국제(CC BY 4.0) 라이선스](https://creativecommons.org/licenses/by/4.0/)에 따라 무료로 자유롭게 이용 가능하도록 라이선스가 부여되었습니다.

이 노트와 [클라임랩(climlab) 소프트웨어](https://github.com/climlab/climlab)의 개발은 브라이언 로즈에게 수여된 AGS-1455071 지원금을 통해 미국 국립과학재단(National Science Foundation)의 부분적인 지원을 받습니다. 여기에 표현된 모든 의견, 발견, 결론 또는 권고는 저의 것이며, 미국 국립과학재단의 견해를 반드시 반영하는 것은 아닙니다.